infra/image/system-service/fixnet.sh

```diff
@@ -1,24 +1,62 @@
 #!/bin/bash -eu
 
+function validFQDN()
+{
+    local name="${1}"
+
+    [[ "${name}" =~ [[:space:]] ]] && return 1
+    [[ "${name}" =~ \. ]] || return 1
+    [[ "${name}" =~ \.\. ]] && return 1
+    for i in ${name//./ }; do
+        [[ "${i}" =~ ^[a-z0-9_/]+$ ]] || return 1
+    done
+    [[ "${name}" == "localhost.localdomain" ]] && return 1
+    return 0
+}
+
+function validIPv4()
+{
+    local ip="${1}"
+    local rematch="^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$"
+
+    [[ "${ip}" =~ ${rematch} ]] || return 1
+    for i in ${ip//./ }; do
```

Comment:
cosmetic: 

s/i/octet/g 

_octet_ has a meaning

Reply:
This is a bash syntax to replace "." (dot) with " " (space). ${parameter//pattern/string} replaces in the whole string and ${parameter/pattern/string} only the first one.